### DISCUSSION TITLE
Add custom labels to existing model

### DISCUSSION THREAD

**Discussion by sudarshan-koirala** [explosion/spaCy#8759] [Help: Coding & Implementations] [answered]
Hello, Is it possible to extract the labels on which the Named Entities were trained ? For example, how can I see what labels are being provided for EVENT entity during training ? The trained model does not capture some of the EVENT entities i want it to capture, so I want to train a custom model with the labels which are not already available on the pre-trained model and add it into the pipeline of existing model. Any tips on how to overcome this obstacle or link to the documentation would be helpful. Thanks

**Comment by polm** [comment, marked as answer]
If you're referring to [the English models](https://spacy.io/models/en), the training data is OntoNotes, which we cannot share publicly due to licensing reasons. 

I think you want to extract the training data or the labeled entities from the trained model? That is not possible, making a model from training data is (usually) like making an omelet from eggs, you can't got backwards. (I'm a little confused about what you're asking because the label is EVENT.)

If you want to improve the existing models then check the [FAQ entries](https://github.com/explosion/spaCy/discussions/8226) on catastrophic forgetting.

**Reply by sudarshan-koirala** [reply]
Hi, thanks for the reply.
I am currently using entity ruler to add patterns into the existing model for those keywords that the pre-trained model didn't predict correctly. For example, I want `summer festival` to be recognized as EVENT entities so I am creating training data accordingly.
```
Pattern:
{'label': 'EVENT', 'pattern': 'summer festival'}

Training data:
(
        "who is performing at the summer festival this summer.",
        {"entities": [(25, 40, "EVENT")]},
    )
```
My main question is -> Is it possible to view a list or dictionary of keywords that are used to generate the training data ? For example in this case, word `summer festival`  is used in the pattern to make training data for entity EVENT. It would be easier to check what keywords the dictionary doesn't have so I can make the training data based on that ??

**Reply by polm** [reply]
What you are asking for is not possible.

The training data is not generated from a list of keywords. There are a bunch of sentences from sources like newspapers, and people went through and manually marked locations, or organizations, or the names of people and so on. It's possible a list was used to automate part of this, but it's also possible it was entirely manual. If a list was used for automation there is no record of it.

The NER model also doesn't work by memorizing keywords. It also learns context, so if you say "I took the train to Piffle", it can figure out that's a place even if it's never seen the word "Piffle" before. That's the big difference between rule-based methods and statistical methods like models - the model is not just a list of phrases to match.

Using just the trained model file there is no straightforward way to get a list of everything with a given label in the training data.
